network/p2p/router.go

```diff
@@ -198,6 +199,9 @@ func (r *Router) RegisterAppProtocol(handlerID uint64, handler Handler, nodeSamp
 	}, nil
 }
 
+// AppRequest routes an AppRequest to a Handler based on the handler prefix.
+//
+// The message is dropped if no matching handler can be found.
```

Comment:
It's a bit non-obvious here why `handler.AppRequest` propagates an error as fatal if it returns an error.

It looks like the reason for this is that the error returned from the handler gets dropped here: 

https://github.com/ava-labs/avalanchego/blob/master/network/p2p/handler.go#L99

but a fatal error could be returned from the `AppSender` when trying to send the response to the peer here: 

https://github.com/ava-labs/avalanchego/blob/master/network/p2p/handler.go#L111

Could we either add a comment internally to this function explaining why this is the one case where it looks like we're returning a fatal error from the handler or find a different way to make it more clear?

Reply:
Added a comment